packages/core/contracts/merkle-distributor/implementation/MerkleDistributor.sol

```diff
@@ -142,17 +130,33 @@ contract MerkleDistributor is Ownable, Testable {
         for (uint256 i = 0; i < claims.length; i++) {
             Claim memory claim = claims[i];
             require(claim.account == account, "Invalid account in batch claim");
-            _markClaimed(claim);
+            _verifyAndMarkClaimed(claim);
             amountToClaim = amountToClaim.add(claim.amount);
         }
-        _disburse(IERC20(rewardToken), account, amountToClaim);
+        IERC20(rewardToken).safeTransfer(account, amountToClaim);
```

Comment:
this seems dangerous. dont you also need to verify all claim windows match the rewardToken?

Reply:
ohh yes great point, should add an `if(claim.rewardToken == rewardToken)` to the loop above

Reply:
This is a great point. That is definitely exploitable.

Reply:
We probably need a similar check to `claim.account == account` for `rewardToken`.

Reply:
Great catch, fixed